src/views/docs-view/utils/all-docs-server.ts

```diff
@@ -0,0 +1,240 @@
+		 */
+		const isNotCustomLandingPath = customLandingPaths.every((landingPath) => {
+			const isMatch = entryPath == landingPath || entryPath == `${landingPath}/`
+			return !isMatch
+		})
+		return isNotCustomLandingPath
+	})
+}
+
+/**
+ *
+ * getStaticProps
+ *
+ */
+
+/**
+ * Get static props for an [...allDocs] page
+ */
+async function getStaticProps(
+	{ params }: GetStaticPropsContext,
+	productData: ProductData
+): Promise<GetStaticPropsResult<Record<string, unknown>>> {
+	// Determine which basePath we're working with from incoming params,
+	// and determine the pageParams within that basePath
+	const { rootDocsPath, pageParams } = parseRootDocsPath(
+		params,
+		productData.rootDocsPaths
+	)
+	// Get the configured getStaticProps for the target rootDocsPath
+	const { getStaticProps: generatedGetStaticProps } =
+		getStaticGenerationFunctionsBase({
+			baseName: rootDocsPath.name,
+			basePath: rootDocsPath.path,
+			navDataPrefix: rootDocsPath.navDataPrefix,
+			product: productData,
+			productSlugForLoader: rootDocsPath.productSlugForLoader,
+		})
+	// Note that the context { params } are constructed here, not passed directly.
+	return await generatedGetStaticProps({
+		params: { page: pageParams },
+	})
+}
+
+/**
+ * Given params from [...allDocs] page getStaticProps context,
+ * and an array of rootDocsPaths,
+ * Return the rootDocsPath config that matches the provided params,
+ * and return the pageParams to be passed to our content loader.
+ *
+ * This allows us to use our existing generateGetStaticFunctions,
+ * configuring it with the matched `rootDocsPath`, and passing `pageParams`
+ * which do _not_ include the matched `rootDocsPath`.
+ */
+export function parseRootDocsPath(params, rootDocsPaths) {
+	// Determine which basePath we're working with, from incoming params
+	const [targetBasePath, maybeTargetBasePath, ...restParams] = params.allDocs
+	let basePath
+	let pageParams
+	// Note that some basePaths are nested, such as "cloud-docs/agents"
+	// For these paths, we need to massage params even further to get the
+	// pageParams we can pass to the usual docs getStaticProps loader
+	const maybeNestedBasePath = [targetBasePath, maybeTargetBasePath].join('/')
+	const allBasePaths = rootDocsPaths.map(({ path }) => path)
+	const isNestedBasePath = allBasePaths.indexOf(maybeNestedBasePath) !== -1
+	if (isNestedBasePath) {
+		basePath = maybeNestedBasePath
+		pageParams = [...restParams]
+	} else {
+		basePath = targetBasePath
+		pageParams =
+			typeof maybeTargetBasePath !== 'undefined'
+				? [maybeTargetBasePath, ...restParams]
+				: [...restParams]
+	}
+	// Get the configured getStaticProps for the target rootDocsPath
+	const rootDocsPath = rootDocsPaths.find((entry) => entry.path == basePath)
+	return { rootDocsPath, pageParams }
+}
+
+/**
+ * Generates static functions for use in an [...allDocs].tsx DocsView page file.
+ */
+export function getStaticGenerationFunctions(productSlug: ProductSlug): {
```

Comment:
This file is where most of the significant changes in this PR live. I've tried to strike a balance here between meeting Terraform's needs, while also looking forward to potentially abstracting further (with other products also using the `[...allDocs]` approach, or perhaps even all product docs rendering from a single `[productSlug]/[...allDocs]` page file, if that's even possible).